### PR TITLE
Make LineQubit and GridQubit immutable

### LINKED ISSUE #2356: Qubits shouldn't be mutable
```python
import cirq
q1 = cirq.LineQubit(1)
circuit = cirq.Circuit(cirq.X(q1))
print(circuit)
q1.x = 2
print(circuit)
```

```
1: ───X───
2: ───X───
```

### PR DESCRIPTION
**Pull request by fedimser**
Coordinates of qubits on line or grid now are properties and can't be changed.

Fixes #2356.

### CODE REVIEW

**Comment by fedimser** [comment]
Hi. I am not sure whether this solution is good enough.
It doesn't make qubits "really" immutable, because I still can access private properies (e.g. qubit._x), and I can add and modfy any other fields.

If this is acceptable solution, then NamedQubit is already good, because `name` is a property.

@mpharrigan @Strilanc

File: cirq/devices/grid_qubit_test.py [hunk 1]
```diff
@@ -185,3 +185,9 @@ def test_to_json():
         'row': 5,
         'col': 6,
     }
+
+
+def test_immutable():
+    with pytest.raises(AttributeError, match="can't set attribute"):
+        q = cirq.GridQubit(1, 2)
+        q.col = 3
```
Comment (cduck) [inline cirq/devices/grid_qubit_test.py]: Also test `.row`.

**Comment by mpharrigan** [comment]
Yes, nothing is truly immutable in python. This is good

**Comment by mpharrigan** [comment]
Or `object.__setattr__()` (which is how frozen dataclasses set their members to begin with :)

File: cirq/devices/grid_qubit_test.py (hunk 1 above)
Comment (fedimser) [reply cirq/devices/grid_qubit_test.py]: Added a test.

**Pull request merged**